flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/ComparableTypeStrategy.java

```diff
@@ -169,8 +173,7 @@ private boolean areDistinctTypesComparable(LogicalType firstType, LogicalType se
 	}
 
 	private boolean areStructuredTypesComparable(LogicalType firstType, LogicalType secondType) {
-		return firstType.equals(secondType) &&
-			hasRequiredComparision((StructuredType) firstType);
+		return firstType.equals(secondType) && hasRequiredComparision((StructuredType) firstType);
 	}
 
 	private boolean areCollectionsComparable(LogicalType firstType, LogicalType secondType) {
```

Comment:
nit: a row is not really a collection, according to SQL standard and LogicalTypeFamily I would rather call this `areConstructedTypesComparable`